mlflow/langchain/_langchain_autolog.py

```diff
@@ -105,53 +106,76 @@ def _update_langchain_model_config(model):
         return True
 
 
-def _inject_mlflow_callback(func_name, mlflow_callback, args, kwargs):
-    if func_name == "invoke":
-        from langchain.schema.runnable.config import RunnableConfig
+def _inject_callbacks(original_callbacks, new_callbacks):
+    """
+    Inject list of callbacks into the original callbacks.
+    For RunnableConfig, the callbacks is defined as List[BaseCallbackHandler] or BaseCallbackManager
+    https://github.com/langchain-ai/langchain/blob/ed980601e1c630f996aabf85df5cb26178e53099/libs/core/langchain_core/callbacks/base.py#L636
+    """
+    from langchain_core.callbacks.base import BaseCallbackManager
 
-        in_args = False
-        # `config` is the second positional argument of runnable.invoke function
-        # https://github.com/langchain-ai/langchain/blob/7d444724d7582386de347fb928619c2243bd0e55/libs/core/langchain_core/runnables/base.py#L468
-        if len(args) >= 2:
-            config = args[1]
-            in_args = True
-        else:
-            config = kwargs.get("config", None)
-        if config is None:
-            callbacks = [mlflow_callback]
-            config = RunnableConfig(callbacks=callbacks)
-        else:
-            callbacks = config.get("callbacks") or []
-            if not isinstance(callbacks, list):
-                callbacks = [callbacks]
-            config["callbacks"] = [*callbacks, mlflow_callback]
-        if in_args:
-            args = (args[0], config) + args[2:]
+    if isinstance(original_callbacks, BaseCallbackManager):
```

Comment:
This is great :) Can we add some information in the Docs FAQ for LangChain about this behavior? (that either a CallbackManager or a subclass of `BaseCallbackHandler` or a `List[BaseCallbackHandler]` are supported?) This feels documentation-worthy

Reply:
I'll do this in a follow-up PR